client.go

```diff
@@ -10,71 +10,64 @@ import (
-		Invoices      *InvoicesService
-		Plans         *PlansService
-		AddOns        *AddOnsService
-		Subscriptions *SubscriptionsService
-		Transactions  *TransactionsService
-	}
-
-	// Params are used to send parameters with the request.
-	Params map[string]interface{}
-)
+const defaultBaseURL = "https://%s.com/"
+
+// Client manages communication with the Recurly API.
+type Client struct {
+	// client is the HTTP Client used to communicate with the API.
+	Client *http.Client
+
+	// subdomain is your account's sub domain used for authentication.
+	subDomain string
+
+	// apiKey is your account's API key used for authentication.
+	apiKey string
+
+	// BaseURL is the base url for api requests.
+	BaseURL string
+
+	// Services used for talking with different parts of the Recurly API
+	Accounts      AccountsService
+	Adjustments   AdjustmentsService
+	Billing       BillingService
+	Coupons       CouponsService
+	Redemptions   RedemptionsService
+	Invoices      InvoicesService
+	Plans         PlansService
+	AddOns        AddOnsService
+	Subscriptions SubscriptionsService
+	Transactions  TransactionsService
+}
 
-// NewClient creates a new Recurly API Client.
-func NewClient(subDomain string, apiKey string, httpClient *http.Client) *Client {
+// NewClient sets the unexported fields on the struct and returns a Client.
+func NewClient(subDomain, apiKey string, httpClient *http.Client) *Client {
 	if httpClient == nil {
 		httpClient = http.DefaultClient
 	}
 
-	c := &Client{
-		client:    httpClient,
+	client := &Client{
+		Client:    httpClient,
 		subDomain: subDomain,
 		apiKey:    apiKey,
 		BaseURL:   fmt.Sprintf(defaultBaseURL, subDomain),
 	}
 
-	c.Accounts = &AccountsService{client: c}
-	c.Adjustments = &AdjustmentsService{client: c}
-	c.Billing = &BillingService{client: c}
-	c.Coupons = &CouponsService{client: c}
-	c.Redemptions = &RedemptionsService{client: c}
-	c.Invoices = &InvoicesService{client: c}
-	c.Plans = &PlansService{client: c}
-	c.AddOns = &AddOnsService{client: c}
-	c.Subscriptions = &SubscriptionsService{client: c}
-	c.Transactions = &TransactionsService{client: c}
-
-	return c
+	client.Accounts = NewAccountsImpl(client)
+	client.Adjustments = NewAdjustmentsImpl(client)
+	client.Billing = NewBillingImpl(client)
+	client.Coupons = NewCouponsImpl(client)
+	client.Redemptions = NewRedemptionsImpl(client)
+	client.Invoices = NewInvoicesImpl(client)
+	client.Plans = NewPlansImpl(client)
+	client.AddOns = NewAddOnsImpl(client)
+	client.Subscriptions = NewSubscriptionsImpl(client)
+	client.Transactions = NewTransactionsImpl(client)
+
+	return client
 }
 
-// newRequest creates an authenticated API request that is ready to send.
-func (c *Client) newRequest(method string, action string, params Params, body interface{}) (*http.Request, error) {
+// NewRequest creates an authenticated API request that is ready to send.
+func (c *Client) NewRequest(method string, action string, params Params, body interface{}) (*http.Request, error) {
```

Comment:
Unexported in [5238e2](https://github.com/blacklightcms/go-recurly/pull/10/commits/322b61900754b8bcf3719fcf55a8b1a7955238e2)